packages/flutter/lib/src/material/expansion_panel.dart

```diff
@@ -121,8 +172,91 @@ class ExpansionPanelList extends StatelessWidget {
   /// The duration of the expansion animation.
   final Duration animationDuration;
 
+  //Whether multiple panels can be open simultaneously
+  final bool _allowMultiplePanelsOpen;
+
+  @override
+  State<StatefulWidget> createState() => new _ExpansionPanelListState();
+}
+
+class _ExpansionPanelListState extends State<ExpansionPanelList> {
+  Map<int, bool> _openPanels = <int, bool>{};
+
+  @override
+  void initState() {
+    super.initState();
+    if (!widget._allowMultiplePanelsOpen) {
+      for (int i = 0; i < widget.children.length; i += 1) {
+        assert(widget.children[i] is ExpansionPanelRadio,
+        'All children of ExpansionPanel.radio need to be of type ExpansionPanelRadio');
+        final ExpansionPanelRadio _widgetChild = widget.children[i];
+        _openPanels[_widgetChild.value] = _widgetChild.initializesExpanded;
+      }
+
+      assert(_debugCheckExpandLimit(),
+      'This expansion panel widget initialized with more than one panel open');
+    }
+  }
+
+  @override
+  void didUpdateWidget(ExpansionPanelList oldWidget) {
+    super.didUpdateWidget(oldWidget);
+
+    if (!widget._allowMultiplePanelsOpen) {
+      final Map<int, bool> newMap = <int, bool>{};
+      for (int i = 0; i < widget.children.length; i += 1) {
+        final ExpansionPanelRadio child = widget.children[i];
+        final int childKey = child.value;
+        newMap[childKey] = _openPanels[childKey] ?? child.initializesExpanded;
+
+        assert(widget.children[i] is ExpansionPanelRadio,
+        'All children of ExpansionPanel.radio need to be of type ExpansionPanelRadio');
+      }
+      _openPanels = newMap;
+      assert(_debugCheckExpandLimit(),
+      'This expansion panel widget initialized with more than one panel open');
+    }
+  }
+
+  // Used for initializing the radio type list.
+  // Checks whether more than one panel is set to expanded when the radio expansion
+  // panel is used.
+  bool _debugCheckExpandLimit() {
+    int openPanels = 0;
+    for (bool value in _openPanels.values) {
+      if (value)
+        openPanels += 1;
+    }
+    return openPanels <= 1;
+  }
+
   bool _isChildExpanded(int index) {
-    return children[index].isExpanded;
+    if (!widget._allowMultiplePanelsOpen) {
+      final ExpansionPanelRadio radioWidget = widget.children[index];
+      return _openPanels[radioWidget.value];
+    }
+    return widget.children[index].isExpanded;
+  }
+
+  void _handlePressed(bool isExpanded, int index) {
+    if (widget.expansionCallback != null)
+      widget.expansionCallback(index, isExpanded);
```

Comment:
Does this callback get invoked twice in radio mode?

Reply:
You mean by the rest of the code below? I don't think so, only possibly one other panel if it was open before.